examples/generative/speaker_recognition_using_cnn.py

```diff
@@ -0,0 +1,520 @@
+
+# Let's separate these 2 categories into 2 folders:
+# - 'audio' which will contain all the speakers' audio folders and files
+# - 'noise' which will contain all the noise sounds and folders
+
+if os.path.exists(DATASET_AUDIO_PATH) is False:
+    os.makedirs(DATASET_AUDIO_PATH)
+
+if os.path.exists(DATASET_NOISE_PATH) is False:
+    os.makedirs(DATASET_NOISE_PATH)
+
+if (
+    len(os.listdir(DATASET_AUDIO_PATH)) == 0
+    and len(os.listdir(DATASET_NOISE_PATH)) == 0
+):
+    for folder in os.listdir(DATASET_ROOT):
+        if os.path.isdir(os.path.join(DATASET_ROOT, folder)):
+            if folder in [AUDIO_SUBFOLDER, NOISE_SUBFOLDER]:
+                continue
+            elif folder in NOISE_FOLDERS:
+                shutil.move(
+                    os.path.join(DATASET_ROOT, folder),
+                    os.path.join(DATASET_NOISE_PATH, folder),
+                )
+            else:
+                shutil.move(
+                    os.path.join(DATASET_ROOT, folder),
+                    os.path.join(DATASET_AUDIO_PATH, folder),
+                )
+
+"""
+## Noise Preparation
+"""
+
+# Let's first get the list of all noise files
+noise_paths, _, _ = tf.python.keras.preprocessing.dataset_utils.index_directory(
+    DATASET_NOISE_PATH, "inferred", FILE_FORMATS, shuffle=False
+)
+
+# In this section:
+# - We load all noise samples which should have been resampled to 16000
+# - We split those noise samples to chuncks of 16000 samples which
+#   correspond to 1 second duration each
+
+i = tf.constant(0)
+
+noises = tf.Variable([])
+
+
+def read_and_process_noise(i, noises):
+    noise, sr = tf.audio.decode_wav(tf.io.read_file(noise_paths[i]), desired_channels=1)
+
+    if sr == DESIRED_SR:
+        # Number of slices of 16000 each that can be generated from the noise sample
+        slices = int(noise.shape[0] / DESIRED_SR)
+
+        # Split the noise sample, stack and concatenate them
+        if noises.shape[0] == 0:
+            noises = tf.stack(tf.split(noise[: slices * DESIRED_SR], slices))
+        else:
+            noises = tf.concat(
+                [noises, tf.stack(tf.split(noise[: slices * DESIRED_SR], slices))],
+                axis=0,
+            )
+    else:
+        print("SR for {} is different than desired. Ignoring it".format(noise_paths[i]))
+
+    return i + 1, noises
+
+
+cond = lambda i, noises: tf.less(i, len(noise_paths))
+
+_, noises = tf.while_loop(cond, read_and_process_noise, [i, noises])
+
+print(
+    "{} noise files were split into {} noise samples where each is {} sec. long".format(
+        len(noise_paths), noises.shape[0], noises.shape[1] // DESIRED_SR
+    )
+)
+
+"""
+## Data Generation
+"""
+
+
+def audio_dataset_from_directory(
+    directory,
+    labels="inferred",
+    label_mode="int",
+    class_names=None,
+    desired_channels=-1,
+    desired_samples=-1,
+    batch_size=32,
+    shuffle=True,
+    seed=None,
+    validation_split=None,
+    subset=None,
+    follow_links=False,
```

Comment:
Please restrict the options to what you will actually use here. This is a code example, not a new API.

This could be a single-use function -- or it could even be in-lined in the outer file.

Reply:
I simplified the functions and reduced the arguments to the minimum needed